samtranslator/model/eventsources/push.py

```diff
@@ -994,6 +995,16 @@ def to_cloudformation(self, **kwargs):
 
     def _inject_lambda_config(self, function, userpool):
         event_triggers = self.Trigger
+
+        if isinstance(event_triggers, dict):
```

Comment:
Could you instead check setting the third parameter of `PropertyType` at line to `False`? (Forbidding intrinsic functions). See what kind of error message we get, if it makes sense for the customers. With this, also check the error that you get if you pass a dict that is not an intrinsic.

If it doesn't and we keep the current code, we should maybe change the `isinstance(event_triggers, dict)` to a `not list or string` test.

Reply:
Setting this parameter returns the following for both intrinsics and generic dicts: 

```
ERROR:__main__:Invalid Serverless Application Specification document. Number of errors found: 1. Resource with id [ImplicitApiFunctionOneTrigger] is invalid. Type of property 'Trigger' is invalid.
ERROR:__main__:["Resource with id [ImplicitApiFunctionOneTrigger] is invalid. Type of property 'Trigger' is invalid."]
```

This seems to make enough sense for implementation so I'll go ahead and do it this way.